extra/mariabackup/wsrep.cc

```diff
@@ -67,21 +70,35 @@ xb_write_galera_info(bool incremental_prepare)
 	long long	seqno;
 	MY_STAT		statinfo;
 
-	/* Do not overwrite existing an existing file to be compatible with
-	servers with older server versions */
-	if (!incremental_prepare &&
-		my_stat(XB_GALERA_INFO_FILENAME, &statinfo, MYF(0)) != NULL) {
-
-		return;
-	}
-
 	xid.null();
 
+	/* try to read last wsrep XID from innodb rsegs, we will use it
+	   instead of galera info file received from donor
+	*/
 	if (!trx_rseg_read_wsrep_checkpoint(xid)) {
-
+		/* no worries yet, SST may have brought in galera info file
+		   from some old MariaDB version, which does not support
+		   wsrep XID storing in innodb rsegs
+		*/
 		return;
 	}
 
+	/* if SST brought in galera info file, copy it as *_SST file
+	   this will not be used, saved just for future reference
+	*/
+	if (my_stat(XB_GALERA_INFO_FILENAME, &statinfo, MYF(0)) != NULL) {
+		FILE* fp_in  = fopen(XB_GALERA_INFO_FILENAME, "r");
+		FILE* fp_out = fopen(XB_GALERA_INFO_FILENAME_SST, "w");
+
+		char buf[BUFSIZ];
+		size_t size;
+		while ((size = fread(buf, 1, BUFSIZ, fp_in))) {
+			fwrite(buf, 1, size, fp_out);
+		}
```

Comment:
No checks for `fp_in` stream status after `while()` loop. If there was an error reading the stream, it will go unnoticed. I'd check here if the EOF was reached (if `feof(fp_in)` returns nonzero.